### PR TITLE
Adding react router, redux example in Home

### PR DESCRIPTION
**Pull request by NateMoeller**

### CODE REVIEW

File: web/package.json [hunk 1]
```diff
@@ -6,7 +6,13 @@
...
     "react": "^16.4.0",
     "react-dom": "^16.4.0",
-    "react-scripts": "1.1.4"
+    "react-redux": "^5.0.7",
+    "react-router": "^4.2.0",
+    "react-router-dom": "^4.2.2",
+    "react-scripts": "1.1.4",
+    "redux": "^4.0.0",
+    "redux-logger": "^3.0.6",
+    "redux-thunk": "^2.3.0"
```
Comment (nckackerman) [inline web/package.json]: alphabetize your imports you god damn neanderthal

File: web/src/app/App.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+class App extends Component {
+
+  render() {
+    return (
+      <Router>
+        <div>
+          <Route exact path='/' component={HomeContainer}/>
+          <Route path='/home'   component={HomeContainer}/>
+          <Route path='/about' component={AboutContainer}/>
```
Comment (nckackerman) [inline web/src/app/App.js]: Weird spacing. There extra spaces after '/home'... lets clean this up.

File: web/src/app/about/AboutContainer.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React, { Component } from 'react';
+import AboutComponent from './AboutComponent';
+
+class AboutContainer extends Component {
+
```
Comment (nckackerman) [inline web/src/app/about/AboutContainer.js]: be consistent with spacing. AboutContainer has a blank line between 
Component {...

render(..

but AboutComponent does not.

File: web/src/app/home/HomeComponent.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class HomeComponent extends Component {
+  render() {
+    return (
+      <div>
+        <div onClick={this.props.fetchApi}>Click me to ping the api endpoint using redux!</div>
+        {this.props.showSpinner ? <div>Fetching Api...</div> : <div>Done.</div>}
+      </div>
+
```
Comment (nckackerman) [inline web/src/app/home/HomeComponent.js]: remove blank line to stay consistent with other code

File: web/src/app/home/HomeComponent.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+
+class HomeComponent extends Component {
```
Comment (nckackerman) [inline web/src/app/home/HomeComponent.js]: Why two lines between imports and component definition? Lets push for consistancy in style across frontend code.

File: web/src/app/home/duck/reducers.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+// reducers.js
```
Comment (nckackerman) [inline web/src/app/home/duck/reducers.js]: dont commit commented out code

File: web/src/index.js [hunk 7]
```diff
@@ -1,8 +1,23 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+
```
Comment (nckackerman) [inline web/src/index.js]: why a blank line here?

File: web/src/index.js [hunk 8]
```diff
@@ -1,8 +1,23 @@
...
+import logger from 'redux-logger';
+import thunk from 'redux-thunk';
+import { createStore, applyMiddleware } from 'redux';
+import rootReducer from './reducers';
```
Comment (nckackerman) [inline web/src/index.js]: alphabetize imports please

**Comment by NateMoeller** [comment]
@nckackerman all good comments. I can look into getting the linter working so this kind of stuff is flagged in the editor.

File: web/package.json (hunk 1 above)
Comment (NateMoeller) [reply web/package.json]: It is?!?

Comment (nckackerman) [reply web/package.json]: Christ... Im an idiot. Nevermind :/ hah

**Comment by NateMoeller** [comment]
Just added a linter to help flag style inconsistencies! Should be good to go now

**Pull request merged**